src/java.base/share/classes/sun/security/ssl/CipherSuite.java

```diff
@@ -857,6 +853,20 @@
 
     final boolean exportable;
 
+    private static final HashMap<Integer, CipherSuite> maps_id = new HashMap<>();
+    private static final HashMap<String, CipherSuite> maps_name = new HashMap<>();
+    private static final CipherSuite[] ciphers = CipherSuite.values();
+
+    static {
+        for(CipherSuite cs : ciphers) {
+            maps_id.put(cs.id, cs);
+            maps_name.put(cs.name, cs);
+            for (String alias : cs.aliases) {
+                maps_name.put(alias, cs);
```

Comment:
A minor concern here. HashMap can't have duplicate keys.  what if there are duplicated names/aliases?

Even it's not the case now, CipherSuite is subject to change in the future and I think it is a good opportunity to detect key duplication here. Currently,  it's silently overwritten. This may introduce inconsistent behavior for nameOf.

Reply:
this could be potentially stored in immutable collections which might be slightly more compact + they throw when they encounter duplicate keys
1) change base type to Map
2) copy ciphers array into Map.Entry array
3) maps_id = Map.ofEntries(entries) // + handle IAE

similar for the name map but with a list in between.

there might be a JDK internal shortcut to get to new ImmutableCollections.MapN<>(flatArray) right away without the Map.Entry step - I am sure a reviewer will chime in if there is.

Reply:
Thank you for the feedback. I have updated to use ImmutableMap. However, I had to use `copyOf(Map)` method instead. 

It would not check for duplicate values. Would it make more sense to add a test case to check for duplicates instead?. If so I can add that, else i'll check for duplicate while creating the map.